modules/apps/frontend-js/frontend-js-spa-web/src/main/resources/META-INF/resources/senna/events/EventEmitter.js

```diff
@@ -0,0 +1,478 @@
+	 * listener is removed for it.
+	 * @param {!(Array|string)} event
+	 * @param {number} amount The amount of times this event should be listened
+	 * to.
+	 * @param {!Function} listener
+	 * @return {!EventHandle} Can be used to remove the listener.
+	 */
+	many(event, amount, listener) {
+		const events = this.toEventsArray_(event);
+		for (let i = 0; i < events.length; i++) {
+			this.many_(events[i], amount, listener);
+		}
+
+		return new EventHandle(this, event, listener);
+	}
+
+	/**
+	 * Adds a listener that will be invoked a fixed number of times for a single
+	 * event. After the event is triggered the specified amount of times, the
+	 * listener is removed.
+	 * @param {string} event
+	 * @param {number} amount The amount of times this event should be listened
+	 * to.
+	 * @param {!Function} listener
+	 * @protected
+	 */
+	many_(event, amount, listener) {
+		const self = this;
+
+		if (amount <= 0) {
+			return;
+		}
+
+		/**
+		 *
+		 */
+		function handlerInternal() {
+			if (--amount === 0) {
+				self.removeListener(event, handlerInternal);
+			}
+			listener.apply(self, arguments); // eslint-disable-line
+		}
+
+		self.addSingleListener_(event, handlerInternal, false, listener);
+	}
+
+	/**
+	 * Checks if a listener object matches the given listener function. To match,
+	 * it needs to either point to that listener or have it as its origin.
+	 * @param {!Object} listenerObj
+	 * @param {!Function} listener
+	 * @return {boolean}
+	 * @protected
+	 */
+	matchesListener_(listenerObj, listener) {
+		const fn = listenerObj.fn || listenerObj;
+
+		return (
+			fn === listener ||
+			(listenerObj.origin && listenerObj.origin === listener) // eslint-disable-line
+		);
+	}
+
+	/**
+	 * Removes a listener for the specified events.
+	 * Caution: changes array indices in the listener array behind the listener.
+	 * @param {!(Array|string)} event
+	 * @param {!Function} listener
+	 * @return {!Object} Returns emitter, so calls can be chained.
+	 */
+	off(event, listener) {
+		this.validateListener_(listener);
+		if (!this.events_) {
+			return this;
+		}
+
+		const events = this.toEventsArray_(event);
+		for (let i = 0; i < events.length; i++) {
+			this.events_[events[i]] = this.removeMatchingListenerObjs_(
+				toArray(this.events_[events[i]]),
+				listener
+			);
+		}
+
+		return this;
+	}
+
+	/**
+	 * Adds a listener to the end of the listeners array for the specified events.
+	 * @param {!(Array|string)} events
+	 * @param {!Function} listener
+	 * @return {!EventHandle} Can be used to remove the listener.
+	 */
+	on() {
+		return this.addListener.apply(this, arguments); // eslint-disable-line
+	}
+
+	/**
+	 * Adds handler that gets triggered when an event is listened to on this
+	 * instance.
+	 * @param {!function()} handler
+	 */
+	onListener(handler) {
+		this.listenerHandlers_ = this.addHandler_(
+			this.listenerHandlers_,
+			handler
+		); // eslint-disable-line
```

Comment:
What's this suppression for? It's not obvious to me. In generally, we should make suppressions specific anyway (ie. they should explicitly mention the rule that is being disabled); same goes for other suppressions in this file.

Reply:
Directly coming from `metal.js`... will take a look and remove, good catch!

Reply:
https://github.com/liferay-frontend/liferay-portal/pull/489#issuecomment-725325366